### ISSUE TITLE
what is the meaning of this line

### ISSUE THREAD

**Issue by 18360865802** [StephanieWyt/RDGCN#2]
The idea of ​​the article is great！ 
I understand the paper but I have a problem when I refer to the source code.Main.py 140 lines
what is  `logits = f_1 + tf.transpose(f_2)`  mean?

**Comment by StephanieWyt** [comment]
Thank you for your interest in our work!
Logits in this line is w^r_{ij}*a^r*[c_i||c_j] in Eq.4 in our paper. This equals to first multiply w^r with c (in_fts in our code), and then multiply in_fts with first half dimensions of a^r and last half dimensions of a^r (f_1 and f_2.transpose respectively), and add them together.

**Comment by 18360865802** [comment]
> Thank you for your interest in our work!
> Logits in this line is w^r_{ij}_a^r_[c_i||c_j] in Eq.4 in our paper. This equals to first multiply w^r with c (in_fts in our code), and then multiply in_fts with first half dimensions of a^r and last half dimensions of a^r (f_1 and f_2.transpose respectively), and add them together.

I don't quite understand why conv1d is used.And the meaning using transpose.
I would be very grateful if the troubles can be answered.

**Comment by StephanieWyt** [comment]
Conv1d with filters=k and kernel_size=1 equals to multiply a matrix W with shape [d, k] (the input layer is of shape n*d).
The transpose and the add with broadcast in line 140 produces a n*n matrix logits. The ith line and jth column of logits is the sum of the ith element in f_1 and the jth element in f_2, meaning the sum of w^r*a^r_1*c_i and w^r*a^r_2*c_j, where a^r=[a^r_1||a^r_2] and both a^r_1 and a^r_2 is half as long as a^r.
